From 8e220531577980d445c0ce6f382ec93b51cce3df Mon Sep 17 00:00:00 2001
From: Peter Yin <peter.yin@quantatw.com>
Date: Wed, 25 Feb 2026 15:19:35 +0800
Subject: [PATCH 209/209] i3c: master: ast2600: Fix missing of_node for virtual
 I2C adapter

The ASPEED AST2600 I3C master driver creates a virtual I2C adapter to
provide backward compatibility with I2C devices. However, the current
implementation does not associate this virtual adapter with any
Device Tree node.

This patch propagates the of_node from the I3C master platform device
to the virtual I2C adapter's device structure. This ensures that
standard I2C aliases are correctly resolved and bus numbering remains
consistent across reboots.
---
 drivers/i3c/master/ast2600-i3c-master.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/i3c/master/ast2600-i3c-master.c b/drivers/i3c/master/ast2600-i3c-master.c
index a0a48f10bccb..90f2b203f5e3 100644
--- a/drivers/i3c/master/ast2600-i3c-master.c
+++ b/drivers/i3c/master/ast2600-i3c-master.c
@@ -294,6 +294,7 @@ static int ast2600_i3c_probe(struct platform_device *pdev)
 
 	i3c->dw.platform_ops = &ast2600_i3c_ops;
 	i3c->dw.ibi_capable = true;
+	i3c->dw.base.i2c.dev.of_node = np;
 	return dw_i3c_common_probe(&i3c->dw, pdev);
 }
 
-- 
2.43.0

